### ISSUE TITLE
REST Client should auto add open tracing if present

### ISSUE THREAD

**Issue by kenfinnigan** [quarkusio/quarkus#629]
Currently if both `quarkus-opentracing` and `quarkus-rest-client` are present, tracing is not automatically passed through any client calls. The developer needs to explicitly set the provider onto the client

`quarkus-rest-client` should detect the presence of `quarkus-opentracing` and register the appropriate tracing provider.

**Comment by machi1990** [comment]
I wanted to give this a short and noticed that tracing is passed to client calls in the following image.
<img width="703" alt="Screenshot 2019-04-06 at 14 00 02" src="https://user-images.githubusercontent.com/10106536/55669324-23b59a00-5876-11e9-8190-9252f183d8d7.png">
 
I did not understand what more needs to be done. 

So in this commit [ef0a44b9f380d943ef5](https://github.com/machi1990/quarkus/commit/ef0a44b9f380d943ef5cd0ed0065cca002cd1999), I tried to automatically detect `quarkus-opentracing` in `quarkus-rest-client` and adding the tracing provider as suggested by [smallrye-opentracing](https://github.com/smallrye/smallrye-opentracing#microprofile-rest-client). But doing so did not have any "visual" effect as the display in the UI did not change. Am I doing something wrong?
